Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferProxy.h

```diff
@@ -314,13 +316,12 @@ class RemoteImageBufferProxy : public WebCore::ConcreteImageBuffer<BackendType>
         if (UNLIKELY(!m_remoteRenderingBackendProxy))
             return false;
 
-        if (!m_needsFlush)
+        if (!m_hasPendingUpdates)
             return hasPendingFlush();
-        
+        m_hasPendingUpdates = false;
         m_sentFlushIdentifier = WebCore::GraphicsContextFlushIdentifier::generate();
         LOG_WITH_STREAM(SharedDisplayLists, stream << "RemoteImageBufferProxy " << m_renderingResourceIdentifier << " flushDrawingContextAsync - flush " << m_sentFlushIdentifier);
         m_remoteDisplayList.flushContext(m_sentFlushIdentifier);
-        setNeedsFlush(false);
```

Comment:
Is there a reason for moving this setting above. The code is more readable if we flushContext() and then we setNeedsFlush(false) not the other way around.